File: src/ChatApp/Services/user-service/src/main/java/com/chatapp/userservice/entity/RefreshToken.java
package com.chatapp.userservice.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.sql.Timestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Table(name = "refresh_tokens")
public class RefreshToken {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;

    @Column(name = "token", columnDefinition = "VARCHAR(255)", nullable = false)
    String token;

    @Column(name = "expires_at", columnDefinition = "TIMESTAMP", nullable = false)
    Timestamp expiresAt;

    @ManyToOne()
    @JoinColumn(name = "user_id")
    User user;


}

File: src/ChatApp/Services/Message/ChatApp.Message/Models/RefreshToken.cs
using System;
using System.Collections.Generic;

namespace ChatApp.Message.Models;

public sealed partial class RefreshToken
{
    public int Id { get; set; }

    public DateTime ExpiresAt { get; set; }

    public string Token { get; set; } = null!;

    public int? UserId { get; set; }

    public User? User { get; set; }
}
